macros/src/codegen.rs

```diff
@@ -500,8 +502,10 @@ pub fn generate_code(sm: &ParsedStateMachine) -> proc_macro2::TokenStream {
                                     quote!{
                                        #action_code
                                        let out_state = #states_type_name::#out_state;
+                                       let in_state = #states_type_name::#in_state;
```

Comment:
Does this actually work for `in_state` that has data associated with it? I'm not sure how we've moving the internal state machine data around.

Reply:
Nope, the sample does not compile.  Nor do the tests. I'll rework this so it will all work and compile as intended. I think we do not want the data to be passed to the transition callback, since it will make things way more complex (if not impossible for a generic callback).